pkg/sbproxy/reconcile_brokers.go

```diff
@@ -181,6 +185,28 @@ func (r ReconcileBrokersTask) deleteBrokerRegistration(broker *platform.ServiceB
 	}
 }
 
+func (r ReconcileBrokersTask) enableServiceAccessVisibilities(broker *platform.ServiceBroker) {
+	if f, isEnabler := r.platformClient.(platform.ServiceAccess); isEnabler {
+		emptyContext := emptyContext()
+		logrus.WithFields(logBroker(broker)).Info("ReconcileBrokersTask task attempting to enable service access for broker...")
+
+		catalog := broker.Catalog
+		if catalog == nil {
+			logrus.WithFields(logBroker(broker)).Error("Error enabling service access due to missing catalog details")
```

Comment:
Should this then trigger a resync?

Reply:
I am not sure if I understand correctly, but here is a thought:

brokers were just fetched (< 1sec ago) from the Service Manager, along with their catalogs (catalogs are mandatory within the service manager in order for a broker to be registered). So basically if we have no catalog here,  I don't think that any resync(telling SM gimme the broker again but also refetch the catalog) can help us - because we would have hit a code bug here. Let me know if this is the type of "resync" you mean